### PR TITLE
New cinder exporter metrics

### PR DESCRIPTION
**Pull request by vdlobanov**
added new cinder-exporter metrics:
volume_type_used_gb
snapshots_count_by_volume_type
snapshots_count

### CODE REVIEW

File: exporters/cinder.go [hunk 1]
```diff
@@ -308,7 +312,52 @@ func ListVolumeLimits(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metr
 		if err != nil {
 			return err
 		}
-
+		jsonLimits := quotasets.GetUsage(exporter.Client, p.ID).PrettyPrintJSON()
```
Comment (niedbalski) [inline exporters/cinder.go:315]: Can you move this entire logic into its own unit/method so you can add a test for this? I think it is too much logic for a single function.

File: exporters/cinder.go [hunk 2]
```diff
@@ -308,7 +312,52 @@ func ListVolumeLimits(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metr
...
+		var jsonParse map[string]interface{}
+		_ = json.Unmarshal([]byte(jsonLimits), &jsonParse)
+		quota := jsonParse["quota_set"].(map[string]interface{})
```
Comment (niedbalski) [inline exporters/cinder.go:318]: quota, ok := 
if !ok { return fmt.Errorf() }

File: exporters/cinder.go [hunk 3]
```diff
@@ -308,7 +312,52 @@ func ListVolumeLimits(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metr
...
+				if words[0] == "gigabytes" {
+					if wordsLen == 2 {
+						disksQuota := quota[i].(map[string]interface{})
+						inUseGb := disksQuota["in_use"].(float64)
+						volType := words[1]
+						ch <- prometheus.MustNewConstMetric(exporter.Metrics["volume_type_used_gb"].Metric,
+							prometheus.GaugeValue, inUseGb, p.Name, p.ID, volType)
+					} else if wordsLen > 2 {
+						jsonHead := strings.Join(words, "_")
+						disksQuota := quota[jsonHead].(map[string]interface{})
```
Comment (niedbalski) [inline exporters/cinder.go:332]: casting may fail and break the exporting

File: exporters/cinder.go [hunk 4]
```diff
@@ -308,7 +312,52 @@ func ListVolumeLimits(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metr
...
+						inUseGb := disksQuota["in_use"].(float64)
```
Comment (niedbalski) [inline exporters/cinder.go:333]: casting may fail and break the exporting

**Review by niedbalski — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution!. Please refactor the method a bit so that we can test this logic independently. Also, generally assert type conversions for failure and check for existence of keys in the dictionaries.

File: exporters/cinder.go (hunk 1 above)
Comment (vdlobanov) [reply exporters/cinder.go]: Thank you for reviewing!
Moved logic to function in 085a1ae20af7403bcf732e4dc0d2089c129747f3

File: exporters/cinder.go (hunk 2 above)
Comment (vdlobanov) [reply exporters/cinder.go]: fixed in 085a1ae20af7403bcf732e4dc0d2089c129747f3

File: exporters/cinder.go (hunk 3 above)
Comment (vdlobanov) [reply exporters/cinder.go]: fixed in 085a1ae20af7403bcf732e4dc0d2089c129747f3

File: exporters/cinder.go (hunk 4 above)
Comment (vdlobanov) [reply exporters/cinder.go]: fixed in 085a1ae20af7403bcf732e4dc0d2089c129747f3

**Review by niedbalski — CHANGES_REQUESTED** [review]
Hello,

Can you please add unit tests for the function you are adding?

Thanks!

**Pull request closed without merging**